gatsby-node.js

```diff
@@ -0,0 +1,188 @@
+const path = require("path")
+
+exports.onCreatePage = async ({ page, actions }) => {
+  const { createPage, deletePage } = actions
+  deletePage(page)
+  createPage({
+    ...page,
+    context: {
+      ...page.context,
+      sourcePath: path.relative(__dirname, page.componentPath),
+    },
+  })
+}
```

Comment:
It'd be good have comments here, I got no idea why when we create a page, then delete it only to re-create it from reading this  other than to add sourcePath

After looking further down this page, I think all of these could do with some comments explaining what all these changes are for - because I'm a bit stumped